src/components/spreadsheet/utils/equipment-table-utils.ts

```diff
@@ -0,0 +1,280 @@
+                gridContext.dynamicValidation,
+                colDef
+            );
+            if (!isConditionFulfiled) {
+                return false;
+            }
+        } else if (colDef.numeric) {
+            return false;
+        }
+    }
+    if (colDef.crossValidation?.allowZero && fieldVal === 0) {
+        return true;
+    }
+    const minExpression = colDef.crossValidation?.minExpression;
+    const maxExpression = colDef.crossValidation?.maxExpression;
+    if (maxExpression || minExpression) {
+        const minVal = !isNaN(minExpression)
+            ? minExpression
+            : gridContext.dynamicValidation[minExpression];
+        const maxVal = !isNaN(maxExpression)
+            ? maxExpression
+            : gridContext.dynamicValidation[maxExpression];
+        return (
+            (minVal === undefined || fieldVal >= minVal) &&
+            (maxVal === undefined || fieldVal <= maxVal)
+        );
+    }
+    return true;
+};
+
+export const checkValidationsAndRefreshCells = (
+    gridApi: any,
+    gridContext: any
+) => {
+    const updatedErrors = { ...gridContext.editErrors };
+    const columnsWithCrossValidation = gridApi
+        .getColumnDefs()
+        .filter((colDef: any) => colDef.editable);
+    columnsWithCrossValidation.forEach((colDef: any) => {
+        const fieldVal = deepFindValue(
+            gridContext.dynamicValidation,
+            colDef.field
+        );
+        const isValid = isValueValid(fieldVal, colDef, gridContext);
+
+        if (isValid) {
+            delete updatedErrors[colDef.field];
+            gridContext.editErrors = updatedErrors;
+        } else {
+            updatedErrors[colDef.field] = true;
+            gridContext.editErrors = updatedErrors;
+        }
+    });
+    const rowNode = gridApi.getPinnedTopRow(0);
+    if (rowNode) {
+        const refreshConfig = {
+            rowNodes: [rowNode],
+            columns: [
+                ...columnsWithCrossValidation.map(
+                    (colDef: any) => colDef.field
+                ),
+                EDIT_COLUMN,
+            ],
+            force: true,
+        };
+        gridApi.refreshCells(refreshConfig);
+    }
+};
+
+const checkCrossValidationRequiredOn = (
+    dynamicValidation: any,
+    colDef: any
+) => {
+    let dependencyValue = deepFindValue(
+        dynamicValidation,
+        colDef.crossValidation.requiredOn.dependencyColumn
+    );
+    if (typeof dependencyValue === 'boolean') {
+        dependencyValue = dependencyValue ? 1 : 0;
+    }
+    if ('columnValue' in colDef.crossValidation.requiredOn) {
+        // if the prop columnValue exist, then we compare its value with the current value
+        return (
+            dependencyValue !== colDef.crossValidation.requiredOn.columnValue
+        );
+    } else {
+        // otherwise, we just check if there is a current value
+        return dependencyValue === undefined;
+    }
+};
```

Comment:
```suggestion
// you can create those types and interfaces
type DynamicValidation = Record<string, number | undefined>;

interface CrossValidationOptions {
    optional?: boolean;
    requiredOn?: {
        dependencyColumn?: string;
        columnValue?: string;
    };
    allowZero?: boolean;
    minExpression?: number | string;
    maxExpression?: number | string;
}

// you can use it in other locations in this file or other files..
interface CustomColDef extends ColDef {
    crossValidation: CrossValidationOptions;
}

const checkCrossValidationRequiredOn = (
    dynamicValidation: DynamicValidation,
    colDef: CustomColDef
) => {
    const requiredOn = colDef.crossValidation.requiredOn || {};
    let dependencyValue = deepFindValue(
        dynamicValidation,
        requiredOn?.dependencyColumn
    );
    if (typeof dependencyValue === 'boolean') {
        dependencyValue = dependencyValue ? 1 : 0;
    }
    if ('columnValue' in requiredOn) {
        // if the prop columnValue exist, then we compare its value with the current value
        return dependencyValue !== requiredOn.columnValue;
    } else {
        // otherwise, we just check if there is a current value
        return dependencyValue === undefined;
    }
};
```